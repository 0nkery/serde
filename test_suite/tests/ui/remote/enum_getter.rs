#[macro_use]
extern crate serde_derive;

mod remote {
    pub enum E {
        A { a: u8 },
    }
}

#[derive(Serialize)]
#[serde(remote = "remote::E")]
pub enum E {
    A {
        #[serde(getter = "get_a")]
        a: u8,
    },
}

fn main() {}
