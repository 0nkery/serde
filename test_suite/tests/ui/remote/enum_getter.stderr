error: #[serde(getter = "...")] is not allowed in an enum
  --> $DIR/enum_getter.rs:11:1
   |
11 | / #[serde(remote = "remote::E")]
12 | | pub enum E {
13 | |     A {
14 | |         #[serde(getter = "get_a")]
15 | |         a: u8,
16 | |     },
17 | | }
   | |_^

error: aborting due to previous error

